### DISCUSSION TITLE
Any way to track the progress when uploading a big file with ObjectStore::put_multipart?

### DISCUSSION THREAD

**Discussion by bbigras** [apache/arrow-rs#5117] [Q&A] [answered]
https://docs.rs/object_store/latest/object_store/index.html#multipart-upload

**Comment by tustvold** [comment, marked as answer]
You could wrap the AsyncWrite in an impl that keeps track of how many bytes are written. This will be an over-estimate as it will count in-progress uploads but may be good enough.

Otherwise you will need to use the lower level MultipartStore API
